src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int amountInStore = FruitStorage.fruitRepository.get(fruit);

```

Comment:
There is a potential `NullPointerException` here if `FruitStorage.fruitRepository.get(fruit)` returns `null` because the fruit is not present in the repository. You should check if the fruit exists in the storage before attempting to retrieve the amount.